parallel-install/pkg/git/clone.go

```diff
@@ -23,46 +25,65 @@ func (rc *remoteRepoCloner) Clone(url, path string, autoCheckout bool) (*git.Rep
 	})
 }
 
-var defaultCloner repoCloner = &remoteRepoCloner{}
+type RepoCloner interface {
```

Comment:
Why introduce this interface and why make it exported? Exported members change the package's public API.

Reply:
I think its nicer to see which methods are expected of the struct. Although there will probably never be another repoCloner the interface makes it more flexible to include another implementation (see factory NewRepo())

Reply:
My response to this is more or less the same as for the `Repo` struct. Only makes things harder to use and understand with no benefit.